### PR TITLE
[SYCL] [L0] Enable immediate command lists by default.

### PR DESCRIPTION
**Pull request by rdeodhar**

### CODE REVIEW

**Review by smaslov-intel — APPROVED** [review]

File: sycl/test-e2e/DiscardEvents/discard_events_l0_inorder.cpp [hunk 1]
```diff
@@ -2,10 +2,10 @@
...
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
 //
-// RUN: env SYCL_PI_LEVEL_ZERO_BATCH_SIZE=0 ONEAPI_DEVICE_SELECTOR="level_zero:*" %GPU_RUN_PLACEHOLDER %t.out
-// RUN: env SYCL_PI_LEVEL_ZERO_BATCH_SIZE=1 ONEAPI_DEVICE_SELECTOR="level_zero:*" %GPU_RUN_PLACEHOLDER %t.out
-// RUN: env SYCL_PI_LEVEL_ZERO_BATCH_SIZE=2 ONEAPI_DEVICE_SELECTOR="level_zero:*" %GPU_RUN_PLACEHOLDER %t.out
-// RUN: env SYCL_PI_LEVEL_ZERO_BATCH_SIZE=3 ONEAPI_DEVICE_SELECTOR="level_zero:*" %GPU_RUN_PLACEHOLDER %t.out
+// RUN: env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=0 SYCL_PI_LEVEL_ZERO_BATCH_SIZE=0 ONEAPI_DEVICE_SELECTOR="level_zero:*" %GPU_RUN_PLACEHOLDER %t.out
+// RUN: env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=0 SYCL_PI_LEVEL_ZERO_BATCH_SIZE=1 ONEAPI_DEVICE_SELECTOR="level_zero:*" %GPU_RUN_PLACEHOLDER %t.out
+// RUN: env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=0 SYCL_PI_LEVEL_ZERO_BATCH_SIZE=2 ONEAPI_DEVICE_SELECTOR="level_zero:*" %GPU_RUN_PLACEHOLDER %t.out
+// RUN: env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=0 SYCL_PI_LEVEL_ZERO_BATCH_SIZE=3 ONEAPI_DEVICE_SELECTOR="level_zero:*" %GPU_RUN_PLACEHOLDER %t.out
```
Comment (againull) [inline sycl/test-e2e/DiscardEvents/discard_events_l0_inorder.cpp:8]: Fix we discussed offline is merged: https://github.com/intel/llvm/commit/88e7c558ef74164c679f39c3c0a8b61906526d77

Does this test still fail?

**Review by aelovikov-intel — APPROVED** [review]
All the files are de-facto owned by the @intel/dpcpp-l0-pi-reviewers , @smaslov-intel LGTM should be enough. I'm giving my formal approval, but please see of CODEOWNERS file needs to be adjusted.

File: sycl/test-e2e/DiscardEvents/discard_events_l0_inorder.cpp (hunk 1 above)
Comment (rdeodhar) [reply sycl/test-e2e/DiscardEvents/discard_events_l0_inorder.cpp:8]: No, but because the test depends on batching behavior which is only applicable with standard command lists, we have to explicitly set immediate command lists off.

Comment (againull) [reply sycl/test-e2e/DiscardEvents/discard_events_l0_inorder.cpp:8]: Oh, I see, thanks a lot!

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.cpp [hunk 2]
```diff
@@ -1269,6 +1205,76 @@ ur_result_t _ur_device_handle_t::initialize(int SubSubDeviceOrdinal,
     ZeEventsScope = DeviceEventsSetting;
   }
 
+  uint32_t numQueueGroups = 0;
```
Comment (smaslov-intel) [inline sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.cpp:1208]: why did you move this code? is it changed in any way?

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 3]
```diff
@@ -6042,7 +6042,7 @@ pi_result _pi_queue::synchronize() {
     // Otherwise sync all L0 queues/immediate command-lists.
     for (auto &QueueMap : {ComputeQueueGroupsByTID, CopyQueueGroupsByTID}) {
       for (auto &QueueGroup : QueueMap) {
-        if (Device->ImmCommandListUsed) {
+        if (UsingImmCmdLists) {
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/pi_level_zero.cpp:6045]: why is this change needed?

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.cpp (hunk 2 above)
Comment (rdeodhar) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.cpp:1208]: The two blocks are unchanged but rearranged with respect to each other. The old upper part leads to a query for isPVC(), to set limits for copy engines. At that time the device properties are not initialized. By rearranging the blocks the call to isPVC() works as expected. Otherwise there are references to uninitilaized values and a crash.

File: sycl/plugins/level_zero/pi_level_zero.cpp (hunk 3 above)
Comment (rdeodhar) [reply sycl/plugins/level_zero/pi_level_zero.cpp:6045]: Everywhere else we simply use the Queue's value UsingImmCmdLists. This was the one instance where an additional indirection was done. The code is equivalent. I changed it for consistency.
